### ISSUE TITLE
Strange code because of inline comment in expression body

### ISSUE THREAD

**Issue by TWiStErRob** [pinterest/ktlint#2426]
## Expected Behavior
```kotlin
  private fun index(i: Int, j: Int, offset: Int): Int =
    // #define I(_i, _j) ((_j)+ 4*(_i))
    offset + j + 4 * i
```
or
```kotlin
  // #define I(_i, _j) ((_j)+ 4*(_i))
  private fun index(i: Int, j: Int, offset: Int): Int = offset + j + 4 * i

```

## Observed Behavior
```kotlin
  private fun index(i: Int, j: Int, offset: Int): Int = // #define I(_i, _j) ((_j)+ 4*(_i))
    offset + j + 4 * i
```
This is a "technically correct" formatting of an expression body according to "default", but the comment makes it very strange. It separates the documentation from the code it documents.

## Steps to Reproduce
```kotlin
  private fun index(i: Int, j: Int, offset: Int): Int =
    // #define I(_i, _j) ((_j)+ 4*(_i))
    offset + j + 4 * i
```

## Your Environment

* Version of ktlint used: 1.0.1
* Relevant parts of the `.editorconfig` settings: `max_line_length = 140`
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): ktlint CLI
* Version of Gradle used (if applicable): N/A
* Operating System and version: Windows 10

**Comment by paul-dingemans** [comment]
I cannot reproduce this.

```
  private fun index(i: Int, j: Int, offset: Int): Int =
    // #define I(_i, _j) ((_j)+ 4*(_i))
    offset + j + 4 * i
```
with `.editorconfig`:
```
root = true

[*.{kt,kts}]
ktlint_code_style = ktlint_official
ktlint_standard=enabled
# ktlint_experimental = enabled
```
with ktlint `1.0.1` results in:
```
private fun index(
    i: Int,
    j: Int,
    offset: Int,
): Int =
    // #define I(_i, _j) ((_j)+ 4*(_i))
    offset + j + 4 * i
```

Setting `max_line_length` to `140` is not necessary as this is the default for `ktlint_official`.

**Comment by TWiStErRob** [comment]
Let me double check.

**Comment by TWiStErRob** [comment]
> Setting max_line_length to 140 is not necessary as this is the default for ktlint_official.

Thanks for the clue, this lead me to find what I was missing this from OP.

```editorconfig
root = true

[*]
max_line_length = 140

[*.{kt, kts}]
ktlint_code_style = intellij_idea
```

Sorry, when writing I noticed that max_line_length was necessary to repro, but didn't notice the style difference.

**Comment by paul-dingemans** [comment]
I can now reproduce it. Code style `intellij_idea` is the only code style for which `max_line_length` is not set by default.

The result below:
```
  private fun index(i: Int, j: Int, offset: Int): Int = // #define I(_i, _j) ((_j)+ 4*(_i))
    offset + j + 4 * i
```
is actually correct for your configuration and is according to the [kotlin style guide](https://kotlinlang.org/docs/coding-conventions.html#expression-bodies). The comment is part of the expression body, and the first line of the expression body does fit on the same line as the function signature.
> If the function has an expression body whose first line doesn't fit on the same line as the declaration, put the = sign on the first line and indent the expression body by four spaces.

This particular behavior can be changed with `.editorconfig` property `ktlint_function_signature_body_expression_wrapping`. https://pinterest.github.io/ktlint/latest/rules/configuration-ktlint/#wrapping-the-expression-body-of-a-function The default value of this property is in sync the Kotlin Coding Conventions.
